trajopt_ifopt/src/joint_acceleration_constraint.cpp

```diff
@@ -0,0 +1,111 @@
+/**
+ * @file joint_acceleration_constraint.h
+ * @brief The joint_acceleration constraint
+ *
+ * @author Ben Greenberg
+ * @date April 22, 2021
+ * @version TODO
+ * @bug No known bugs
+ *
+ * @copyright Copyright (c) 2021, Southwest Research Institute
+ *
+ * @par License
+ * Software License Agreement (Apache License)
+ * @par
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * @par
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <trajopt_ifopt/constraints/joint_acceleration_constraint.h>
+
+TRAJOPT_IGNORE_WARNINGS_PUSH
+#include <console_bridge/console.h>
+TRAJOPT_IGNORE_WARNINGS_POP
+
+namespace trajopt
+{
+JointAccelConstraint::JointAccelConstraint(const Eigen::VectorXd& targets,
+                                           std::vector<JointPosition::ConstPtr> position_vars,
+                                           const std::string& name)
+  : ifopt::ConstraintSet(static_cast<int>(targets.size()) * static_cast<int>(position_vars.size() - 1), name)
```

Comment:
I think this should be `position_vars.size - 2`?

Reply:
I agree, it seems like that should be a `- 2`, but when I run with that the cost gets enormous, whereas when I run with a `- 1` the cost behaves as expected. I'm looking to see if there are more places aside from here and line 57 where this change needs to be propagated, but I haven't had luck finding it yet. Any thoughts why it seems to behave well with the `- 1`?

Reply:
Not really. Maybe whatever problem you are testing it on has a high acceleration on the last point? You might try making a unit test that defines the variables to be some known function and then make sure that getValue returns the correct value. For instance, make position_vars follow a cubic and make sure the values of this constraint increase linearly